internal/controller/secrets.go

```diff
@@ -94,3 +100,30 @@ func (secrets *Secrets) WriteK8sSecret(ctx context.Context, keyData OnboardRespo
 	}
 	return nil
 }
+
+// NewClientSet returns a new Kubernetes clientset based on the execution environment.
+func NewClientSet() (*kubernetes.Clientset, error) {
```

Comment:
Why is this function in a file named `secrets.go`?

Reply:
renamed to authenticator